consumer/template_group_writer.go

```diff
@@ -0,0 +1,159 @@
+package consumer
+
+import (
+	"context"
+	"fmt"
+	"github.com/goat-project/goat/consumer/wrapper"
+	"os"
+	"path"
+	"text/template"
+)
+
+const (
+	templateFileExtension = "tmpl"
+	filenameFormat        = "%014d"
+)
+
+type vmsTemplateData struct {
+	Vms []interface{}
+}
+
+// TemplateGroupWriter converts each record to template and writes it to file. Multiple records may be written into a single file.
+type TemplateGroupWriter struct {
+	dir          string
+	templatesDir string
+	countPerFile uint64
+	templateName string
+	template     *template.Template
+	records      []interface{}
+}
+
+// NewTemplateGroupWriter creates a new TemplateGroupWriter.
+func NewTemplateGroupWriter(dir, templatesDir, templateName string, countPerFile uint64) TemplateGroupWriter {
+	return TemplateGroupWriter{
+		dir:          dir,
+		templatesDir: templatesDir,
+		countPerFile: countPerFile,
+		templateName: templateName,
+		template:     nil,
+		records:      make([]interface{}, countPerFile),
```

Comment:
This looks strange to me. What will happen when the last batch that doesn't have all `countPerFile` records gets rendered? I would guess that either it will end with panic since `record[x]` will be `nil` or it will create some "empty" records. Or am I wrong?

Reply:
You're right, this was previously solved with the countInFile variable. I'll see what I can do with this.

Reply:
yes, the array is copied to array of correct size and the template should process the array correctly